api/v1beta1/common_types.go

```diff
@@ -22,9 +22,9 @@ import (
 )
 
 const (
-	// ClusterLabelName is the label set on machines linked to a cluster and
+	// ClusterNameLabel is the label set on machines linked to a cluster and
 	// external objects(bootstrap and infrastructure providers).
-	ClusterLabelName = "cluster.x-k8s.io/cluster-name"
+	ClusterNameLabel = "cluster.x-k8s.io/cluster-name"
```

Comment:
The renames should be eventually all added to the migration doc.

The migration doc should be created by the release team in 1-2 weeks: https://github.com/kubernetes-sigs/cluster-api/blob/d146f6d9e264241c5db29e8ecf99a390ab97807d/docs/release/release-tasks.md#add-docs-to-collect-release-notes-for-users-and-migration-notes-for-provider-implementers

Theoratically we could do it here already but I would like to give the first release team a chance to try out our new release process, ...

Reply:
Let's also in general hold merging this PR until at least after the v1.3.0 release (1st December) just to ensure it doesn't interfere with the release / cherry-picks until then.

Reply:
Sure, will make sure the PR is on hold until at least after the v1.3.0 release.